src/server.mjs

```diff
@@ -1094,17 +1109,139 @@ api.post("/getPathNames", (req, res, next) => {
   });
 });
 
-api.post("/getBedRegions", (req, res) => {
+// returns whether or not a string is a valid http url
+function isValidURL(string) {
+  if (!string) {
+    return False;
+  }
+
+  let url;
+
+  try {
+    url = new URL(string)
+  } catch(error) {
+    return false;
+  }
+
+  return url.protocol === "http:" || url.protocol === "https:";
+}
+
+// given the datapath and url, download file to the datapath
+const downloadFile = async(fileURL, dataPath, fileName) => {
+
+  if (!fileName) {
+    return;
+  }
+
+  const options = {
+    method: "GET",
+    credentials: "omit",
+    cache: "default",
+    signal: timeoutController(15).signal // pass in abort controller for timeout
+  };
+
+  const response = await fetch(fileURL, options);
+  
+  const destinationPath = path.resolve(dataPath, fileName);
+
+  const fileStream = fs.createWriteStream(destinationPath, { flags: 'wx' });
+  await finished(Readable.fromWeb(response.body).pipe(fileStream));
+
+}
+
+// download chunks specified by bed URL
+const retrieveChunks = async(bedURL, chunks) => {
+  // up go a directory level in the url
+  const dataPath = bedURL.split('/').slice(0, -1).join('/');
+  for (const dirName of chunks) {
+    const chunkDir = config.tempDirPath + "/" + dirName;
+    // create directory for the chunk
+    if (!fs.existsSync(chunkDir)) {
+      fs.mkdirSync(chunkDir, { recursive: true });
+    } else {
+      continue;
+    }
+
+    // fetch and download files and put them in the directory
+    const options = {
+      method: "GET",
+      credentials: "omit",
+      cache: "default",
+      signal: timeoutController(15).signal // pass in abort controller for timeout
+    };
+
+
+
+    // retrieve and download each file in the chunk directory
+
+    let chunkContentURL = dataPath + "/" + dirName + "/chunk_contents.txt";
+    let response = await fetch(chunkContentURL, options);
+
+    const chunkContent = await response.text();
```

Comment:
It looks like there's no limit on the size of the chunk contents file that we will accept. It could be 30 gigabytes and we will run out of memory to store it and crash.